src/pybamm/solvers/solution.py

```diff
@@ -460,9 +460,20 @@ def first_state(self):
         else:
             all_yps = self.all_yps[0][:, :1]
 
+        # check if all_ys is empty (i.e. if IDAKLU solver + output_variables used)
+        if isinstance(self.all_ys[-1], casadi.DM):
+            empty_ys = False
+        else:
+            empty_ys = self.all_ys[0].shape[0] == 0
+
         new_sol = Solution(
             self.all_ts[0][:1],
-            self.all_ys[0][:, :1],
+            # Get first state from initial conditions if all_ys is empty
+            (
+                self.all_ys[0][:, :1]
+                if not empty_ys
+                else self.all_models[0].y0full.reshape(-1, 1)
+            ),
```

Comment:
The pre-commit formatting makes this a bit hard to read. Can we move this into an if statement above?